src/adapters/vertexai/utils.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright 2024 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { isString } from "remeda";
+import { Serializer } from "@/serializer/serializer.js";
+import { VertexAI, GenerativeModel } from "@google-cloud/vertexai";
+import { getPropStrict, getProp } from "@/internals/helpers/object.js";
+import { GenerateContentResponse } from "@google-cloud/vertexai";
+
+export function processContentResponse(response: GenerateContentResponse): string {
+  if (response.candidates != undefined) {
+    return response.candidates
+      .map((candidate) => candidate.content.parts.map((part) => part.text!))
+      .flat()
+      .join();
+  }
+  return "Empty";
+}
+
+export function registerGenerativeModel() {
+  Serializer.register(GenerativeModel, {
+    toPlain: (value) => ({
+      project: getPropStrict(value, "project"),
+      location: getPropStrict(value, "location"),
+      model: getPropStrict(value, "model"),
+      safetySettings: getPropStrict(value, "safetySettings"),
+      generationConfig: getPropStrict(value, "generationConfig"),
+      systemInstruction: getProp(value, ["systemInstruction"]),
+    }),
+    fromPlain: (value) => {
+      const vertexAI = new VertexAI({ project: value.project, location: value.location });
+      return vertexAI.getGenerativeModel({
+        model: value.model,
+        safetySettings: value.safetySettings,
+        generationConfig: value.generationConfig,
+      });
+    },
+  });
+}
+
+export function updateClient(
+  client: GenerativeModel,
+  schema: string | Record<string, any>,
+): GenerativeModel {
+  const project = getPropStrict(client, "project");
+  const location = getPropStrict(client, "location");
+  const model = getPropStrict(client, "model");
+  const schemaJson = isString(schema) ? JSON.parse(schema) : schema;
+  const modelParams = {
+    model: model,
+    generationConfig: { responseSchema: schemaJson, responseMimeType: "application/json" },
+  };
+  const vertexAI = new VertexAI({ project: project, location: location });
+  return vertexAI.getGenerativeModel(modelParams);
```

Comment:
Is the creation of a new client really needed? They do not provide a way how to override some parameters for a single request.

Reply:
I was looking for the way but I couldn't find it.  Here is the api reference of the client (GenerativeModel) https://cloud.google.com/vertex-ai/generative-ai/docs/reference/nodejs/latest/vertexai/generativemodel?hl=en&authuser=1

Reply:
Yes, that's make sense.  I'll make changes.

Reply:
It's done now.